pkgs/development/python-modules/pymavlink/default.nix

```
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, fetchPypi, future, lxml }:
+
+buildPythonPackage rec {
+  pname = "pymavlink";
+  version = "2.4.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0y9rz3piddzdjpp7d5w9xi6lc9v9b4p4375a5hrfiphrmhds85i3";
+  };
+
+  propagatedBuildInputs = [ future lxml ];
```

future is a python2 only library
```suggestion
  propagatedBuildInputs = [ lxml ] ++ lib.optionals isPy27 [ future ];
```

---

`future` is an unconditional `install_requires`, therefore it is still needed on Python 3 to satisfy dependency resolution.

---

this is technically incorrect (it should only be needed for python2), but it's an upstream issue. All good to leave it in propagatedBuildInputs, I think the closure size addition is small anyway.